plugins/talk-plugin-notifications/server/NotificationManager.js

```diff
@@ -149,17 +77,170 @@ class NotificationManager {
           .map(({ category }) => category)
           .join(', ')}] handlers when the '${event}' event is emitted`
       );
-      broker.on(event, this.handle(handlers));
+      broker.on(event, this.handleUserEvent(handlers));
     });
   }
 
   /**
-   * handle will wrap a notification handler and attach it to the notification
-   * stream system.
+   * registerDigests will register the digest handlers.
+   *
+   * @param {Array<Object>} handlers digest handlers for options related to digesting
+   */
+  registerDigests(...handlers) {
+    this.digests.push(...handlers);
+  }
+
+  /**
+   * startDigesting will register all the digests to run and setup the cron
+   * jobs.
+   */
+  startDigesting() {
+    this.digests.forEach(({ frequency, config }) => {
+      new CronJob(
+        merge(config, {
+          start: true,
+          onTick: this.handleDigestEvent(frequency),
+        })
+      );
+    });
+  }
+
+  handleDigestEvent(frequency) {
+    return async () => {
+      // Create a system context to send down.
+      const ctx = this.context.forSystem();
+
+      try {
+        // Pull out some useful tools.
+        const {
+          connectors: { models: { User }, services: { I18n: { t } } },
+        } = ctx;
+
+        const organizationName = await getOrganizationName(ctx);
+        if (!organizationName) {
+          ctx.log.error(
+            'could not send the notification, organization name not in settings'
+          );
+          return;
+        }
+
+        const subject = t(
+          'talk-plugin-notifications.templates.digest.subject',
+          organizationName
+        );
+
+        // Continue to pull from the Users digest until the queue is empty.
+        while (true) {
+          // Pull notifications from a user that have notifications enabled for
+          // `frequency` and currently have notifications.
+          const user = await User.findOneAndUpdate(
+            {
+              'metadata.notifications.settings.digestFrequency': frequency,
+              'metadata.notifications.digests': { $exists: true, $ne: [] },
+            },
+            { $set: { 'metadata.notifications.digests': [] } }
+          );
+          if (!user) {
+            // There are no more users that meet the search criteria! We're
+            // done!
+            ctx.log.info('no notifications from database');
+            break;
+          }
+
+          // Begin rendering the user's digest.
+          const digests = get(user, 'metadata.notifications.digests');
+          if (!digests) {
+            // We couldn't get the digest from the user (even after Mongo said
+            // we would get it?).
```

Comment:
Did it happen to you?

Reply:
(more of defensive programming than anything, because we're accessing such a deeply nested object)